src/test/lua/exasol/adapter/databricks/ConnectionReader_spec.lua

```diff
@@ -23,88 +23,278 @@ local function read_address(address, user, password)
+* Specify one of the supported AuthMech values 3 (token auth) or 11 (M2M OAuth).]]
+            }, {
+                name = "unsupported AuthMech",
+                connection = {address = "jdbc:databricks://host:123;AuthMech=unsupported"},
+                expected_error = [[E-VSDAB-22: Connection 'my_connection' contains JDBC URL 'jdbc:databricks://host:123;AuthMech=unsupported' with unsupported AuthMech property value 'unsupported'.
 
 Mitigations:
 
-* Only token authentication is supported, please specify USER='token' and PASSWORD='<token>` in the connection.]])
-    end)
-    it("fails for missing password", function()
-        assert.has_error(function()
-            read_connection({address = "jdbc:databricks://example.com:123", user = "token"})
-        end, [[E-VSDAB-14: Connection 'my_connection' does not contain a valid token.
+* Use one of the supported AuthMech values 3 (token auth) or 11 (M2M OAuth).]]
+            }, {
+                name = "unsupported numeric AuthMech",
+                connection = {address = "jdbc:databricks://host:123;AuthMech=4"},
+                expected_error = [[E-VSDAB-22: Connection 'my_connection' contains JDBC URL 'jdbc:databricks://host:123;AuthMech=4' with unsupported AuthMech property value '4'.
+
+Mitigations:
+
+* Use one of the supported AuthMech values 3 (token auth) or 11 (M2M OAuth).]]
+            }, {
+                name = "token auth: missing user",
+                connection = {address = "jdbc:databricks://host:123;AuthMech=3"},
+                expected_error = [[E-VSDAB-13: Connection 'my_connection' contains invalid user <missing value>.
+
+Mitigations:
+
+* Only token authentication is supported, please specify USER='token' and PASSWORD='<token>` in the connection.]]
+            }, {
+                name = "token auth: invalid user",
+                connection = {address = "jdbc:databricks://host:123;AuthMech=3", user = "wrong_user"},
+                expected_error = [[E-VSDAB-13: Connection 'my_connection' contains invalid user 'wrong_user'.
+
+Mitigations:
+
+* Only token authentication is supported, please specify USER='token' and PASSWORD='<token>` in the connection.]]
+            }, {
+                name = "token auth: missing password",
+                connection = {address = "jdbc:databricks://host:123;AuthMech=3", user = "token", password = nil},
+                expected_error = [[E-VSDAB-14: Connection 'my_connection' does not contain a valid token.
+
+Mitigations:
+
+* Please specify PASSWORD='<token>` in the connection.]]
+            }, {
+                name = "m2m oauth: non-empty user",
+                connection = {address = "jdbc:databricks://host:123;AuthMech=11", user = "non-empty", password = nil},
+                expected_error = [[E-VSDAB-23: Connection 'my_connection' uses M2M OAuth but 'USER' or 'IDENTIFIED BY' fields are not empty.
+
+Mitigations:
+
+* Specify Client ID and Client Secret as 'OAuth2ClientId' and 'OAuth2Secret' in JDBC URL.]]
```

Comment:
The mitigation looks strange for this case.
Shouldn't it be something like:
```suggestion
Mitigations:

* Use empty username or choose another authentication method.]]
```
?